extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -276,18 +276,29 @@ private Uni<TokenVerificationResult> verifyCodeFlowAccessTokenUni(RoutingContext
                 && (resolvedContext.oidcConfig.authentication.verifyAccessToken
                         || resolvedContext.oidcConfig.roles.source.orElse(null) == Source.accesstoken)) {
             final String codeAccessToken = (String) vertxContext.get(OidcConstants.ACCESS_TOKEN_VALUE);
-            return verifyTokenUni(resolvedContext, codeAccessToken);
+            return verifyTokenUni(resolvedContext, codeAccessToken, null);
         } else {
             return NULL_CODE_ACCESS_TOKEN_UNI;
         }
     }
 
-    private Uni<TokenVerificationResult> verifyTokenUni(TenantConfigContext resolvedContext, String token) {
+    private Uni<TokenVerificationResult> verifyTokenUni(TenantConfigContext resolvedContext, String token, UserInfo userInfo) {
         if (OidcUtils.isOpaqueToken(token)) {
             if (!resolvedContext.oidcConfig.token.allowOpaqueTokenIntrospection) {
                 LOG.debug("Token is opaque but the opaque token introspection is not allowed");
                 throw new AuthenticationFailedException();
             }
+            // verify opaque access token with UserInfo if enabled and introspection URI is absent
+            if (resolvedContext.oidcConfig.token.verifyOpaqueAccessTokenWithUserInfo
```

Comment:
Please check no introspection endpoint is available

Reply:
@sberyozkin I think next line is sufficient.

Reply:
`resolvedContext.provider.getMetadata().getIntrospectionUri() == null`

Reply:
Oops, missed it :-)